interview-questions.md

```diff
@@ -7,62 +7,67 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 INSTRUCTOR EXAMPLE: What is a conditional statement?
 
-  Your answer:
+  Your answer: A code block that is checking to see if a certain condtion is true or false in order to execute code.
 
-  Researched answer:
+  Researched answer: conditionals perform different computations or actions depending on whether a programmer-defined boolean condition evaluates to true or false.
 
 
 
 1. What is git? What is the difference between git and Github?
 
-  Your answer:
+  Your answer: Git is an open source tool used to manage source code on your local machine. GitHub is a website where developers/organizations can upload source code and projects. Github integrates with git allowing users to upload and pull code that is hosted on GitHub.
 
   Researched answer:
+  Git:Git A distributed version control system for tracking changes in source code during software development
+
+  GitHub: A provider of Internet hosting for software development and version control using Git.
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: In JS the '&&' as well as '||' (Logical AND, Logical OR return a boolean.
 
-  Researched answer:
+  Researched answer: In JavaScript, both || and && are logical short-circuit operators that return the first fully-determined “logical value” when evaluated from left to right. 
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: Index refers to the position of a given element in an array starting from the zero index going from left to right. Value refers to actual value at the given element. 
 
-  Researched answer:
+  Researched answer: 
+  Index: used as a reference to the actual place holder in the element's value of an array.
+  Value: 
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: In a loop iteration refers to the process of iterating or looping through as long as a condtion is met. 
 
-  Researched answer:
+  Researched answer: Iteration refers to the process in which the code segment is executed once. One iteration refers to 1-time execution of a loop. A loop can undergo many iterations.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: In proper pair programming there is a driver who has hands on the keyboard and a navigator. A navigator does not type during the process and is reviewing in real time.
+ 
+  Researched answer: Pair programming is an agile software development technique in which two programmers work together at one workstation. One, the driver, writes code while the other, the observer or navigator, reviews each line of code as it is typed in.
```

Comment:
Please make sure that all of your researched answers are _**formed in your own words**_